### DISCUSSION TITLE
Workspace and .twinproj files

### DISCUSSION THREAD

**Discussion by KarlSevenSeven** [twinbasic/twinbasic#526] [General]
VB6 has a project, and the files in use are listed in the VBP.
Means all the single files are separate, they can also reside in different folders.

In tB, we have one .code-workspace file and one .twinproj file as a pair.
As far as I understand, a .twinproj holds ALL source files together.
The .twinproj is referenced in the .code-workspace file.

It is not possible to copy a pair of files, rename them both and it just works.
Q:
What is the recommended way to copy a complete project?

My opinion:
I like it a lot better to have a bunch of single, individual files.
I can search through then by just using Windows Explorer.
Edit them in a text editor.
Make individual copies etc. etc..

What do you think?

**Comment by bclothier** [comment]
Just FYI:

> The twinproj format does allow us to optionally support external files;  we just haven't done some of the additional work in this area to enable it.  See #62    
>
> Once external files are supported, the twinproj file can basically be used like a traditional project file (like vbp) OR a full all-in-one solution.  The choice will be yours.

_Originally posted by @WaynePhillipsEA in https://github.com/WaynePhillipsEA/twinbasic/issues/35#issuecomment-874933094_

**Comment by KarlSevenSeven** [comment]
Aha, good.
Perhaps that comes automatically with 'Open VBP'.

**Reply by WaynePhillipsEA** [reply]
Yup, that's the plan

**Comment by WaynePhillipsEA** [comment]
Also, if/when we move to Theia (#458), we'll be able to (optionally) support an all-in-one package of just a twinproj file (i.e. no code-workspace file needed), which would help simplify.
